src/tree.rs

```diff
@@ -188,23 +190,25 @@ fn _bipartition_tree(
 ) -> Vec<(usize, Vec<usize>)> {
     let mut pops = pops;
     let spanning_tree_graph = &spanning_tree.graph;
-    let mut same_partition_tracker: Vec<Vec<usize>> =
-        vec![vec![]; spanning_tree_graph.node_bound()]; // Keeps track of all all the nodes on the same side of the partition
+    let mut same_partition_tracker: Vec<LinkedList<usize>> =
+        vec![LinkedList::new(); spanning_tree_graph.node_bound()]; // Keeps track of all all the nodes on the same side of the partition
```

Comment:
I don't think it is worth using LinkedLists in this case, even more given the fact you need to convert them back to a vector at the end

Reply:
Fair, I just perf tested it, and you're right.